### ISSUE TITLE
newline in guarded match arm should put the `{` on a newline too

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#495]
```
             hir::PatEnum(..) | hir::PatIdent(..) | hir::PatQPath(..)
-                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) =>
-            {
+                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) => {
                 let def = cx.tcx.def_map.borrow().get(&self.pat.id).unwrap().full_def();
```

Probably should be an option. We do a similar thing for where clauses in function decls.

**Comment by nrc** [comment]
In the example, we should do what the original does.

**Comment by topecongiro** [comment]
Closed via #1778.
